lib/fog/vsphere/models/compute/cdrom.rb

```diff
@@ -0,0 +1,66 @@
+module Fog
+  module Compute
+    class Vsphere
+      class Cdrom < Fog::Model
+        identity :key
+
+        attribute :filename
+        attribute :name, :default => "CDROM Drive"
+        attribute :server_id
+        attribute :controller_key
+        attribute :unit_number
+        attribute :start_connected
+        attribute :allow_guest_control
+        attribute :connected
+
+        has_one_identity :server, :servers
+
+        def to_s
+          name
+        end
+
+        def destroy
+          requires :server_id, :key, :unit_number
+
+          service.destroy_vm_cdrom(self)
+          true
+        end
+
+        def save
+          requires :server_id
+
+          if unit_number.nil?
+            used_unit_numbers = server.cdroms.map { |cdrom| cdrom.unit_number }
+            max_unit_number = used_unit_numbers.max
+
+            if max_unit_number > server.cdroms.size
+              # If the max ID exceeds the number of cdroms, there must be a hole in the range. Find a hole and use it.
+              self.unit_number = max_unit_number.times.to_a.find { |i| used_unit_numbers.exclude?(i) }
+            else
+              self.unit_number = max_unit_number + 1
+            end
+          else
+            if server.cdroms.any? { |cdrom| cdrom.unit_number == self.unit_number && cdrom.id != self.id }
+              raise "A cdrom already exists with that unit_number, so we can't save the new cdrom"
+            end
+          end
+
+          data = service.add_vm_cdrom(self)
+
+          if data['task_state'] == 'success'
+            # We have to query vSphere to get the cdrom attributes since the task handle doesn't include that info.
+            created = server.cdroms.all.find { |cdrom| cdrom.unit_number == self.unit_number }
```

Comment:
I guess you could use the `cdrom` collection `#get` here and avoid doing a search in ruby.

```ruby
created = server.cdroms.get(self.unit_number)
```